src/macros.jl

```diff
@@ -625,12 +625,102 @@ julia> @with df @byrow :x * :y
     global scope, then a variable cannot be assigned without using the `global` keyword.
     If the parent scope is a local scope (inside a function or let block for example),
     the `global` keyword is not needed to assign to that parent scope.
+
+!!! note
+    Using `AsTable` inside `@with` block is currently not supported.
 """
 macro with(d, body)
     esc(with_helper(d, body))
 end
 
+astable_rhs_orderby_docs = """
+In operations, it is also allowed to use `AsTable(cols)` to work with
+multiple columns at once, where the columns are grouped together in a
+`NamedTuple`. When `AsTable(cols)` appears in a operation, no
+other columns may be referenced in the block.
+
+Using `AsTable` in this way is useful for working with many columns
+at once programmatically. For example, to order rows by the
+sum of the columns `:a`, `:b`, and `:c`, write
+
+```
+@byrow sum(AsTable([:a, :b, :c]))
+```
+
+This constructs the pair
+
+```
+AsTable(AsTable([:a, :b, :c])) => ByRow(sum)
+```
+
+`AsTable` on the right-hand side also allows the use of the special
+column selectors `Not`, `Between`, and regular expressions. For example,
+to order all rows by the product of all columns starting with `"a"`, write
+
+```
+@byrow prod(AsTable(r"^a"))
+```
+"""
+
+astable_rhs_subset_docs = """
+In operations, it is also allowed to use `AsTable(cols)` to work with
+multiple columns at once, where the columns are grouped together in a
+`NamedTuple`. When `AsTable(cols)` appears in a operation, no
+other columns may be referenced in the block.
+
+Using `AsTable` in this way is useful for working with many columns
+at once programmatically. For example, to select rows where the
+sum of the columns `:a`, `:b`, and `:c` is greater than `5`, write
+
+```
+@byrow sum(AsTable([:a, :b, :c])) > 5
+```
+
+This constructs the pair
+
+```
+AsTable([:a, :b, :c]) => ByRow(t -> sum(t) > 5)
```

Comment:
out of curiosity - and for the future - we  could think what syntax could be used in such cases to create:
```
AsTable([:a, :b, :c]) => ByRow((t -> sum(t) > 5)∘collect)
```

Reply:
I doubt it. No way of knowing how the function uses the named tuple and I don't want to special-case the `sum` as a syntax.